src/github-handler/commit-and-push-handler.ts

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Changes, GitCreateTreeParamsTree, Logger, RepoDomain} from '../types';
+import {Octokit} from '@octokit/rest';
+
+/**
+ * Generate and return a GitHub tree object structure
+ * containing the target change data
+ * See https://developer.github.com/v3/git/trees/#tree-object
+ * @param {Changes} changes the set of repository changes
+ * @returns {GitCreateTreeParamsTree[]} The new GitHub changes
+ */
+function genTreeObj(changes: Changes): GitCreateTreeParamsTree[] {
```

Comment:
nit: generally, we don't want to truncate words
```suggestion
function generateTreeObjects(changes: Changes): GitCreateTreeParamsTree[] {
```